frontend/src/App.svelte

```diff
@@ -275,80 +277,146 @@ const displayInfoMessage= () =>{ //After 1 minute of no nodes recieved, a messag
           enableOptimize: true,
         },
       ],
+    },
+  };
+
+  const handle_close_request = () => {
+    console.log("received in app");
+    show_settings = !show_settings;
+  };
+
+  const handleKeyDown = (e: KeyboardEvent): void => {
+    appKeyMap[e.key] = e.type == "keydown";
+
+    if (
+      (appKeyMap["ArrowRight"] ||
+        appKeyMap["ArrowLeft"] ||
+        appKeyMap["ArrowUp"] ||
+        appKeyMap["ArrowDown"]) &&
+      nonInteractiveState
+    ) {
+      nonInteractiveState = false;
+    }
+
+    // Hide all panels
+    if (appKeyMap["h"] || appKeyMap["H"]) {
+      if (show_legend) toggleLegend();
+
+      if (show_settings) toggleSettings();
+
+      if (show_filter_panel) toggleFilterPanel();
+
+      if (show_timebar) updateTimebar();
+    }
+
+    // Legend panel
+    if (appKeyMap["l"] || appKeyMap["L"]) {
+      toggleLegend();
+    }
+
+    // Options panel
+    if (appKeyMap["s"] || appKeyMap["S"]) {
+      toggleSettings();
+    }
+
+    // Non-interactive mode
+    if (appKeyMap["n"] || appKeyMap["N"]) {
+      toggleInteractiveMode();
+    }
+
+    // Timebar
+    if (appKeyMap["t"] || appKeyMap["T"]) {
+      updateTimebar();
+    }
+
+    // Filter panel (waiting for implementation of toggle filter panel)
+    if (appKeyMap["f"] || appKeyMap["F"]) {
+      toggleFilterPanel();
+    }
+
+    // WIP: Shortcut view
```

Comment:
Is this still WIP?

Reply:
Shouldn't be.

Reply:
I see! 
But since shortcut view is in the settings, is it necessary to have a shortcut to open the shortcut? Isn't it enough to just have the shortcut for the settings instead?

Reply:
Not necessary, no. But it was part of the original acceptance criteria.

Reply:
Yes, but acceptance criteria were not planning on the fact that we would have this setting panel =) 
So I would just remove that from the PR personally, being able to open the setting panel is enough I would say

Reply:
Alright, fixing.